src/sentry/eventstore/processing/multiredis.py

```diff
@@ -0,0 +1,54 @@
+from datetime import timedelta
+from typing import TypeVar
+
+from rediscluster import RedisCluster
+
+from sentry.utils.codecs import JSONCodec
+from sentry.utils.kvstore.encoding import KVStorageCodecWrapper
+from sentry.utils.kvstore.redis import RedisKVStorage
+from sentry.utils.redis import redis_clusters
+
+from .base import EventProcessingStore
+
+T = TypeVar("T", str, bytes)
+
+
+class MultiRedisProcessingStore(EventProcessingStore):
+    """
+    Adapter to shift traffic from one redis cluster to another
+    """
+
+    def __init__(self, **options):
+        inner = MultiRedisKVStorage(
+            old_cluster=redis_clusters.get(options["old_cluster"]),
+            new_cluster=redis_clusters.get(options["new_cluster"]),
+        )
+        super().__init__(KVStorageCodecWrapper(inner, JSONCodec()))
+
+
+class MultiRedisKVStorage(RedisKVStorage[T]):
+    def __init__(self, old_cluster: RedisCluster, new_cluster: RedisCluster) -> None:
+        self.old_cluster = old_cluster
+        self.new_cluster = new_cluster
+
+    def get(self, key: str) -> T | None:
+        str_key = key.encode("utf8")
+        new_val = self.new_cluster.get(str_key)
+        if new_val is not None:
+            return new_val
+        return self.old_cluster.get(str_key)
```

Comment:
In the interest of having a few more options during the incident, I think we should have options (which are runtime) to switch modes. If we wanted to fallback we would have to apply a settings change and restart. 

What about these: 
2 options per cluster: read and write. 
- If they are both on we write on the new and double read 
- if we want to rollout the new one we turn read on both and then write on the new and disable write on the old 
- if we want to rollback we turn write on on the old and turn write off on the new, then after a while we turn off double reads. 
- if we want to cordon the old one entirely we turn read and write on the old one off 
- same if we want to shut down the new one. 

We can make them rollout option `(random() < options.get("......")`